### ISSUE TITLE
Add transient interface

### ISSUE THREAD

**Issue by orgtre** [orgtre/anki-editor#13]
I'm working on a [transient](https://github.com/magit/transient) user interface to anki-editor and am opening this issue to collect feedback and suggestions. It would resolve issues like https://github.com/louietan/anki-editor/issues/79 and beyond. An alternative would be a [hydra](https://github.com/abo-abo/hydra), but I am already familiar with transient and think it is more flexible.

**Comment by orgtre** [comment]
Okay, the first version of the transient is out. Feel free to give it a try and come with suggestions for how the anki-editor workflow can be made more smooth. @cmirdesouza @doolio

The transient is in a separate module. In my setup with straight.el I load it like this:
```elisp
(use-package anki-editor-ui
  :defer t
  :after anki-editor
  :straight (:host github :repo "louietan/anki-editor" :fork "orgtre")
  :bind (:map anki-editor-mode-map
	 ("C-c e" . anki-editor-ui)))
```

**Comment by cmirdesouza** [comment]
I really liked the idea, i was thinking on how the org-agenda works and i didn't think about magit, but it seems like a wonderfull feature especially for newcomers and begginers.

**Comment by orgtre** [comment]
@cmirdesouza Thanks for testing! Indeed I see the same issue appearing randomly sometimes (rarely) also without Doom, but fail to find a way to reproduce it. Basically, the problem seems to be that when typing in a completion-read prompt, sometimes the transient interface pops up again, at which point it eats all the keypresses and it is impossible to input anything into the mini buffer prompt or to get it to disappear. Will try to find a solution.

**Comment by doolio** [comment]
This is a great addition, thank you. Curious why you kept in separate from the main file?

One thing I've noticed when using this UI is that the `:ANKI_FAILURE_REASON:` property remains when successfully pushing a previously failed card. It is removed when pushing such a note not via this UI.

(I'm starting to use this more now so I'll report any other issues I encounter as and when they arise.)

**Comment by orgtre** [comment]
Thanks for testing this and reporting issues! 

I'll try to resolve the issue with `:ANKI_FAILURE_REASON:`. If there are issues with transient which we can't resolve we might considering testing a hydra instead.

I kept it separate to not force transient as a dependency for anki-editor. The transient package is quite big (4000+ lines) so one wouldn't want to load it unless necessary. Also, this seems to be almost standard practice for Emacs packages.

**Comment by doolio** [comment]
> I kept it separate to not force transient as a dependency for anki-editor. The transient package is quite big (4000+ lines) so one wouldn't want to load it unless necessary. Also, this seems to be almost standard practice for Emacs packages.

That's fair. Though isn't `transient` a part of Emacs now or maybe just GNU ELPA.

**Comment by doolio** [comment]
The UI allows one to delete a note but does so only in Anki. Perhaps, an option to delete in Anki and the Org file should be included?

**Comment by orgtre** [comment]
Yes, this option is already there actually: With a universal prefix argument (`C-u`) `anki-editor-delete-note-at-point` will also delete the note from the Org file. This is in the commands docstring but not indicated in the transient interface. I think having at least some indicator in the transient of which commands take interactive arguments, for example like I did [here](https://github.com/orgtre/org-colviewx/blob/main/hydra-dark.png), would be nice.
